recipes/usockets/all/conanfile.py

```diff
@@ -26,17 +28,77 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def _minimum_compilers_version(self, cppstd):
+        standards = {
+            "14": {
+                "Visual Studio": "15",
+                "gcc": "5",
+                "clang": "3.4",
+                "apple-clang": "10",
+            },
+            "17": {
+                "Visual Studio": "16",
+                "gcc": "7",
+                "clang": "6",
+                "apple-clang": "10",
+            },
+        }
+        return standards.get(cppstd) or {}
+
+    @property
+    def _cppstd(self):
+        version = False
+        if self.options.eventloop == "asio":
+            version = "14"
+
+        # OpenSSL wrapper of uSockets uses C++17 features.
+        if self.options.with_ssl == "openssl":
+            version = "17"
+
+        return version
+
+    def validate(self):
+        if self.options.eventloop == "gcd":
+            if self.settings.os != "Linux" or self.settings.compiler != "clang" :
+                raise ConanInvalidConfiguration("gcd eventloop is only supported on Linux with clang")
+
+        if self.options.eventloop == "asio" and tools.Version(self.version) < "0.8.0":
+            raise ConanInvalidConfiguration("asio eventloop is only supported on uSockets >= 0.8.0")
```

Comment:
```suggestion
        if tools.Version(self.version) < "0.8.0" and self.options.eventloop not in ("libuv", "gcd"):
            raise ConanInvalidConfiguration(f"eventloop={self.options.eventloop} is not supported with {self.name}/{self.version}")
```

Reply:
I agree with you.
Your code is more generic than me.

I add "syscall" to "not in" condition.
```
        if tools.Version(self.version) < "0.8.0" and self.options.eventloop not in ("syscall", "libuv", "gcd"):
```